pkg/testutils/echotest/echotest_test.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+//
+
+package echotest
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+)
+
+func TestWalk(t *testing.T) {
+	w := Walk(t, testutils.TestDataPath(t))
+	defer w.Check(t)
```

Comment:
nit: do we need `w.Check()` here? Can `Walk()` simply register a `t.Cleanup()` internally?